### ISSUE TITLE
The `MicroRequest` type for `apollo-server-micro` is not exported correctly

### ISSUE THREAD

**Issue by ChristianIvicevic** [apollographql/apollo-server#6877]
I was attempting to type the factory for a custom context myself as follows:

```ts
import type { MicroRequest } from 'apollo-server-micro';
import { ApolloServer } from 'apollo-server-micro';

const createGraphQLContext = async (args: { req: MicroRequest }) => {
  return { /* ... */ };
};

const apolloServer = new ApolloServer({
  context: createGraphQLContext,
});
```

According [to the docs](https://www.apollographql.com/docs/apollo-server/api/apollo-server/#middleware-specific-context-fields) the type of the argument for the factory should be `{ req: MicroRequest, res: ServerResponse }` however the `MicroRequest` type is not accesible as it is not exported from the root of the package.

**Comment by glasser** [comment]
Out of curiosity, do you actually use micro? Or are you merely using `apollo-server-micro` because a NextJS example inexplicably uses it despite the fact that NextJS is not built on top of Micro?

**Comment by ChristianIvicevic** [comment]
Your second assumption is correct. I saw the use of `apollo-server-micro` documented in the Nexus documentation and was skeptical after following the docs of the transitive dependencies and seeing that `micro` itself has the disclaimer about Vercel in their readme.

An acquaintance guided me to the following PR using `apollo-server-express` that hasn't been merged yet where you commented as well: https://github.com/vercel/next.js/pull/30082

After trying to get it running myself there were different problems with newer versions of Next not allowing for top-level awaits (v12 and SWC seem to have dropped that) and generally missing typings for other parts used within the PR.

**Comment by glasser** [comment]
I guess you can just copy the MicroRequest type from https://github.com/apollographql/apollo-server/blob/main/packages/apollo-server-micro/src/types.ts

We will be releasing Apollo Server 4 soon at which point any Micro integration can be maintained by community members that actually use Micro (although as far as we can tell, nobody actually uses Micro and people who use apollo-server-micro are only using it due to the Next.JS example). Adding more functionality to this dead-end package is not a priority at this time.

Note that AS4 has more explicit APIs for supporting server environments such as Next.JS which don't allow for a standard "startup might fail and you shouldn't start handling requests until after startup has succeeded" flow, so I should be able to replace that PR with one that they hopefully will merge.
